discovery.go

```diff
@@ -245,9 +245,7 @@ func (d *discover) Bootstrap(ctx context.Context, topic string, ready RouterRead
 	}
 
 	t := time.NewTimer(time.Hour)
-	if !t.Stop() {
-		<-t.C
-	}
+	t.Stop()
```

Comment:
please dont change this

Reply:
Out of curiosity, I would like to know why this introduces a bug. It seems to me that the purpose of this code is to create a stopped timer. The line immediately above the call to stop creates a timer with a timeout of 1 hour, so it is not conceivable that the timer's channel should ever need to be cleared.

Further, even if it were possible for more than an hour to elapse between these two consecutive lines of code, reading from the timer's channel would be a bug in go 1.23+, since trying to read from a timer's channel following a call to `Stop` will block forever.

Reply:
read more carefully, if Stop returns false you got to drain it.

Reply:
I did try to read carefully, [Here](https://pkg.go.dev/time@go1.23.0#Timer.Stop) it says this:
> For a chan-based timer created with NewTimer(d), as of Go 1.23, **any receive from t.C after Stop has returned is guaranteed to block rather than receive a stale time value from before the Stop**; if the program has not received from t.C already and the timer is running, Stop is guaranteed to return true. Before Go 1.23, the only safe way to use Stop was insert an extra <-t.C if Stop returned false to drain a potential stale value. See the [NewTimer](https://pkg.go.dev/time@go1.23.0#NewTimer) documentation for more details.

Sorry if I am missing something.

Reply:
I am trying to get a small repro to see if it's actually a golang bug. Ideally `timer.Stop` shouldn't return false. 

The state of the draft PR https://github.com/quic-go/quic-go/pull/4659/files is correct though. You can rely on the capacity of the channel to handle this case.

Reply:
it can (at least before 1.23) return false if the timer has fired.

At any rate, lets make a separate pr foe this and limit this one to the go-libp2p upgrade.

Reply:
>it can (at least before 1.23) return false if the timer has fired.

Yes and it does. 

From go1.23, it shouldn't.